test/framework/src/main/java/org/elasticsearch/test/AbstractWireSerializingTestCase.java

```diff
@@ -48,35 +48,30 @@
     protected abstract Reader<T> instanceReader();
 
     /**
+     * Returns a {@link CopyFunction} that can be used to make an exact copy of
+     * the given instance. This defaults to a function that uses
+     * {@link #copyInstance(Writeable)} to create the copy.
+     */
+    protected CopyFunction<T> getCopyFunction() {
+        return (original) -> copyInstance(original);
+    }
+
+    /**
+     * Returns a {@link MutateFunction} that can be used to make create a copy
+     * of the given instance that is different to this instance. This defaults
+     * to null.
+     */
+    protected MutateFunction<T> getMutateFunction() {
```

Comment:
should this be abstract instead? I think that subclasses should override this, then the test is more complete?